pkg/relayer/session/proof.go

```diff
@@ -218,13 +218,20 @@ func (rs *relayerSessionsManager) proveClaims(
 		// WARNING: Creating a claim and not submitting a proof (if necessary) could lead to a stake burn!!
 		if err != nil {
 			failedProofs = append(failedProofs, sessionTree)
-			rs.logger.Error().Err(err).Msg("failed to determine if proof is required, skipping claim creation")
+			logger.Error().Err(err).Msg("failed to determine if proof is required, skipping claim creation")
 			continue
 		}
 
 		// If a proof is required, add the session to the list of sessions that require a proof.
 		if isProofRequired {
 			sessionTreesWithProofRequired = append(sessionTreesWithProofRequired, sessionTree)
+		} else {
+			rs.removeFromRelayerSessions(sessionTree)
```

Comment:
Optional: See if this shold (could?) return anything and log it.

If/when there's an error, it'll help identify things.